etna/models/base.py

```diff
@@ -563,11 +809,22 @@ def _forecast(self, ts: TSDataset, **kwargs) -> TSDataset:
         horizon = len(ts.df)
         x = ts.to_pandas(flatten=True).drop(["segment"], axis=1)
         # TODO: make it work with prediction intervals and context
-        y = self._base_model.predict(x, **kwargs).reshape(-1, horizon).T
```

Comment:
Maybe we can rewrite this part via decorators: preprocessing and postprocessing via wrapper
Something like:

```
@make_prediction
def _forecast

@make_prediction
def _predict
```